integr8ly/integreatly-operator
# MGDAPI-4122 - Support setting container platform in docker build commands

# Issue link

https://issues.redhat.com/browse/MGDAPI-4122

# What

**ARM64** machines such as M1 MacBooks by default build images in this architecture. When these images are deployed to Openshift which used **AMD64** architecture, pods will fail to run due to mismatch of architectures, with an error similar to:
```
exec /bin/opm: exec format error fix
```

Docker / Podman allows specifying to container build platform for this. By allowing a build flag for this in the image build commands, a consistent image build / architecture would be used across machines for deploying the images onto clusters. This would allow specifying the container build platform to another architecture easier in the future also if there is a use case

* https://docs.docker.com/build/building/multi-platform/#building-multi-platform-images
* https://docs.podman.io/en/latest/markdown/podman-build.1.html#platform-os-arch-variant

The changes were required to install via OLM successfully using an M1 MacBook

# Verification steps



* Verify images are built successfully
```
export ORG=<yourOrg>                                  
```
```
* Verify images are built successfully
# Build Operator Image
OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api make image/build
# Build index images
OC_INSTALL=true ./scripts/bundle-rhmi-operators.sh
# Build Test image
make image/functional/build
# Build OSDE2e image
make image/osde2e/build
# Build scorecard image
make scorecard/build
```
* Verify the index pod deployed is successfully running
```
oc get pods -l "olm.catalogSource"="rhmi-operators" -n openshift-marketplace -o jsonpath="{.items[0].status.phase}"
```
